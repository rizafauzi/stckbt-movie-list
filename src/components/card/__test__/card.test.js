

import React from 'react'
import ReactDOM from 'react-dom'
import Card from '../card.component'
import { cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import renderer from 'react-test-renderer'

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement('div')
  ReactDOM.render(<Card />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('it matches snapshot', () => {
  const dataTest = {
    Title :"Batman v Superman: Dawn of Justice",
    Year :"2016",
    imdbID :"tt2975590",
    Type :"movie",
    Poster :"https://m.media-amazon.com/images/M/MV5BYThjYzcyYzItNTVjNy00NDk0LTgwMWQtYjMwNmNlNWJhMzMyXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
  }
  const tree = renderer.create(<Card data={dataTest} />).toJSON()
  expect(tree).toMatchSnapshot()
})