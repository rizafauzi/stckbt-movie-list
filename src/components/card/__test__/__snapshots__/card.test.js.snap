// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="sc-kEqYlL eYloQa"
  data-testid="card"
>
  <div
    className="sc-bqGHjH RYUa"
  >
    <img
      alt="Logo"
      className="sc-ksluoS fRYgfx"
      src="https://m.media-amazon.com/images/M/MV5BYThjYzcyYzItNTVjNy00NDk0LTgwMWQtYjMwNmNlNWJhMzMyXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
    />
    <div
      className="sc-iqAbSa cjgjuD"
    >
      <button
        className="sc-crzoUp jNUkwQ"
      />
      <div
        className="sc-dIsAE gWpIU"
        onMouseUp={[Function]}
      >
        <h1
          className="sc-bdnylx cmzHvD"
          data-testid="text"
          size={20}
        >
          MOVIE
        </h1>
        <button
          className="sc-gKAblj bmcNPv"
          data-testid="button"
          onMouseDown={[Function]}
          width="150px"
        >
          View Detail
        </button>
      </div>
    </div>
  </div>
  <div>
    <h1
      className="sc-bdnylx cmzHvD"
      data-testid="text"
      size={20}
    >
      Batman v Superman: Dawn of Justice
    </h1>
    <h1
      className="sc-bdnylx fdCiIW"
      data-testid="text"
      size={18}
    >
      2016
    </h1>
  </div>
</div>
`;
